utils/global-state-update-gen/src/generic/state_tracker.rs

```diff
@@ -306,6 +323,14 @@ impl<T: StateReader> StateTracker<T> {
                 );
             }
         }
+
+        if let Some(unbonding_purses) = preexisting_unbonds {
+            if let Some(StoredValue::Unbonding(purses)) =
+                self.entries_to_write.get_mut(&Key::Unbond(account_hash))
+            {
+                purses.extend(unbonding_purses.clone());
+            }
+        }
```

Comment:
Instead of doing this here, I'd make `create_unbonding_purse` read the existing unbonds first, and then append to the existing vector or create a new one if it doesn't exist.

Reply:
Done.